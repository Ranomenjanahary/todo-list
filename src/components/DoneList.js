
export const DoneList = ({ done }) => (
    <div className="done">
      <h3>DONE</h3>
      <ul>
        {done.map((todo, index) => (
          <li key={index}>{todo}</li>
        ))}
      </ul>
    </div>
  );