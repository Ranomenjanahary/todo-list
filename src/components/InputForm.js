import React from 'react';

export const InputForm = ({ handleSubmit, todo, setTodo }) => (
  <form onSubmit={handleSubmit}>
    <input
      type="text"
      value={todo}
      onChange={(e) => setTodo(e.target.value)}
      onKeyDown={(e) => {
        e.key === 'Enter' && handleSubmit();
      }}
      placeholder="Add task"
    />
  </form>
);