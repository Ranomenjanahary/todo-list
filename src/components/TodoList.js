
export const TodoList = ({ doing, handleCheck }) => (
    <div className="todo">
      <h3>TODO</h3>
      <ul>
        {doing.map((todo, index) => (
          <li key={index}>
            {todo}
            <input type="checkbox" onClick={() => handleCheck(index)} />
          </li>
        ))}
      </ul>
    </div>
  );