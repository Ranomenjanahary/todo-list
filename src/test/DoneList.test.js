import React from 'react';
import { DoneList } from '../components/DoneList';
import { render } from '@testing-library/react';

describe('DoneList component', () => {
  it('renders correctly with snapshot', () => {
    const done = ['task 1', 'task 2'];
    const { container } = render(<DoneList done={done} />);
    expect(container).toMatchSnapshot();
  });
});