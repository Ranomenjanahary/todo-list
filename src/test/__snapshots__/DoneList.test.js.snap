// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneList component renders correctly with snapshot 1`] = `
<div>
  <div
    class="done"
  >
    <h3>
      DONE
    </h3>
    <ul>
      <li>
        task 1
      </li>
      <li>
        task 2
      </li>
    </ul>
  </div>
</div>
`;
